ipatests/test_integration/test_cert.py

```diff
@@ -46,3 +47,50 @@ def test_cacert_file_appear_with_option_F(self):
         assert status == "MONITORING"
 
         self.clients[0].run_command(['ls', '-l', '/etc/pki/tls/test.CA'])
+
+
+class TestInstallMasterReplica(IntegrationTest):
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+        tasks.install_replica(self.master, self.replicas[2], setup_ca=True,
+                              setup_kra=False)
+
+    def test_certmomger_tracks_renewed_certs_during_interruptions(self):
+        """Test to check if Certmonger tracks certs in between
+        reboots/interruptions and while in "CA_WORKING" state
+
+        A non-renewal master CA might submit a renewal request before
+        the renewal master actually updating the certs. This is expected.
+
+        But, if certmonger on this node is stopped/interrupted
+        for any reason while the resubmit request is in "CA_WORKING" state
+        it will not be able to track changes to this cert anymore.
+
+        Pagure Issue: https://pagure.io/freeipa/issue/8164
+        """
+        cmd = ['getcert', 'resubmit', '-f',
+               '/var/lib/ipa/ra-agent.pem']
+        result = self.replicas[0].run_command(cmd)
+        # assert 'CA_WORKING' in result.stdout_text
+        request_id = re.findall(r'\d+', result.stdout_text)
+
+        # check if certificate is in CA_WORKING state
+        status = tasks.wait_for_request(self.clients[0], request_id[0], 50)
+        assert status == "CA_WORKING"
+
+        # restart certmonger service on replica
+        self.replicas[0].run_command(['systemctl', 'restart', 'certmonger'])
+        time.sleep(15)
+
+        cmd = ['getcert', 'list', '-f', '/var/lib/ipa/ra-agent.pem']
+        result = self.replicas[0].run_command(cmd)
+        # check if certificate is in SUBMITTING state
+        status = tasks.wait_for_request(self.clients[0], request_id[0], 50)
```

Comment:
see tasks.wait_for_request():
`    for _i in range(0, timeout, 5):
(...)
        if state in ('CA_REJECTED', 'CA_UNREACHABLE', 'CA_UNCONFIGURED',
                     'NEED_GUIDANCE', 'NEED_CA', 'MONITORING'):
            break
        time.sleep(5)
`
I don't think there is any chance of status being SUBMITTING there.

Reply:
I guess I'll just need to assert it then

Reply:
or use a similar algorithm to wait for it, yes.

Reply:
For now I have changed to ```assert``` instead of ```wait_for_request```

Reply:
@rcritten 's patch addresses this